src/main/java/org/commcare/xml/QueryPromptParser.java

```diff
@@ -47,9 +48,21 @@ public QueryPrompt parse() throws InvalidStructureException, IOException, XmlPul
         boolean allowBlankValue = "true".equals(parser.getAttributeValue(null, ATTR_ALLOW_BLANK_VALUE));
         DisplayUnit display = null;
         ItemsetBinding itemsetBinding = null;
-        XPathExpression defaultValue = null;
-
         String defaultValueString = parser.getAttributeValue(null, ATTR_DEFAULT);
+        XPathExpression defaultValue = null;
+        String excludeValueString = parser.getAttributeValue(null, ATTR_EXCLUDE);
+        XPathExpression exclude = null;
+        if (excludeValueString != null) {
+            try {
+                exclude = XPathParseTool.parseXPath(excludeValueString);
+            } catch (XPathSyntaxException e) {
+                InvalidStructureException toThrow = new InvalidStructureException(String.format(
+                        "Invalid XPath Expression in QueryPrompt %s",
+                        e.getMessage()), parser);
+                toThrow.initCause(e);
+                throw toThrow;
+            }
+        }
```

Comment:
This block is almost identical to the one below and could be extracted into a method. I would even go so far as to say it could be added to the parent class since there is similar code in other parses but that could be a separate PR.

```java
XPathExpression exclude = this.parseXPath(excludeValueString, "prompt[@exclude]");

...

private XPathExpression parseXPath(String value, String context) throws InvalidStructureException {
    if (value == null) {
        return null;
    }
    try {
        return XPathParseTool.parseXPath(value);
    } catch (XPathSyntaxException e) {
        InvalidStructureException toThrow = new InvalidStructureException(String.format(
                "Invalid XPath Expression for %s: %s",
                context, e.getMessage()), parser);
        toThrow.initCause(e);
        throw toThrow;
    }
}

Reply:
That makes sense. I can work on that PR after this one.